components/dashboard/src/data/prebuilds/prebuild-logs-emitter.ts

```diff
@@ -68,6 +101,7 @@ export function usePrebuildLogsEmitter(prebuildId: string, taskId: string) {
         );
         return () => {
             controller.abort();
+            emitter.removeAllListeners();
```

Comment:
Overall, the abort/cancellation path in this method is not clear to me.

We are using a multitude of different ways for cleaning things up after ourselves:
 - AbortController
 - Disposable
 - the react EffectCallback result

This feels very grown and ripe for simplification. WDYT?

Reply:
After looking at how this is used, I can see we want to enable two signals from the outside:
 - component unmounted (react hook)
 - stop listening (bc prebuild got aborted, is done, etc.)

Still thing we can make this clearer in the code. Not so sure if it has to be this PR, though, as we had this before. :+1:

Reply:
I agree it'd be great to unify this and get rid of abort controllers or disposables, so they don't need to live in the same places taking care of similar responsibilities. I'd say the problem here is that we rely on so many Disposables and at the same time so many Abort Controllers / signals. I tried to push a commit cleaning this here up a little bit to not use two different techniques for a single task (https://github.com/gitpod-io/gitpod/pull/19887/commits/4124ac152b12c77196b02e398c8c1b1ee364de10).

Reply:
Uh, my comments are gone :confused: Repeating it here anyway:

This is what I see with commit : multiple outputs overlapping
![image](https://github.com/gitpod-io/gitpod/assets/32448529/bf88fd46-7bc6-4eaf-8bf6-0dfcbb2f6325)

Let's leave the disposables/emitter as-is, plus add comments what their purpose is. :pray: 
And sorry for the misleading distraction :bow: